File: javascript/Kobalt_FINPR_197.js
class UserAccount {
  constructor(account_id, balance, active) {
    this.account_id = account_id;
    this.balance = balance;
    this.active = active;
    this.accountTransactions = [];
  }

  deductTransaction(transactions) {
    for (let idx in transactions) {
      // Verify Account Cred.
      if (
        this.account_id === transactions[idx].account_id &&
        this.active &&
        transactions[idx].amount <= this.balance
      ) {
        // Checks how many transactions
        if (this.accountTransactions.length < 2) {
          this.accountTransactions.unshift(transactions[idx]);
          this.balance -= transactions[idx].amount;
        }
        // Compares 3rd item's timestamp to current transaction timestamp
        else if (
          transactions[idx].timestamp - this.accountTransactions[1].timestamp >
          5
        ) {
          this.accountTransactions.unshift(transactions[idx]);
          this.balance -= transactions[idx].amount;
        } else {
          continue;
        }
      }
    }

    return this.accountTransactions;
  }
}

module.exports = UserAccount;

File: python/Kobalt_FINPR_197.py
class UserAccount:
  def __init__(this, account_id, balance, active):
    this.account_id = account_id
    this.balance = balance
    this.active = active
    this.accountTransactions = []

  def deductTransaction(this,transactions):
    print(transactions)
    for transaction in transactions:
        print(transaction)
        if (this.account_id == transaction['account_id'] and this.active and transaction['amount'] <= this.balance):
            if (len(this.accountTransactions)<2):
                this.accountTransactions.insert(0,transaction)
                this.balance -= transaction['amount']
            elif (transaction['timestamp'] - this.accountTransactions[1]['timestamp'] > 5 ) :
                this.accountTransactions.insert(0,transaction)
                this.balance -= transaction['amount']
            else:
              print('Transaction Error')
              return 'Transaction Error'
        print(this.accountTransactions)

                           

# account1 =  UserAccount('Karen', 100, True)

# print(account1.account_id)
# account1.deductTransaction(["apple", "banana", "cherry",{
#   'transaction_id': '1231',
#   'account_id': 'Karen',
#   'amount': 10,
#   'timestamp': 27679677,
# }])
